### DISCUSSION TITLE
beforeinstallprompt doesnt get fired in +layout.svelte

### DISCUSSION THREAD

**Discussion by rheanV** [sveltejs/kit#11550] [Q&A]
I have an installable PWA (for sveltekit app). I want to add a custom button for installing the app. I placed my `window.addEventListener('beforeinstallprompt', (event) => {
    event.preventDefault();
   //logic
  });`  in the onMount of the +layout.svelte file. I have all other configurations correct, meets all installation criteria, and can be installed via the browser download option in the URL bar. However, my "beforeinstallprompt" does not get fired in the +layout.svelte. If I place it in app.html then it is fired but not in +layout.svelte. How should I fix this? What am I missing in my sveltekit app setup?

**Comment by gevera** [comment]
Same issue here. I want to use ```window.addEventListener('beforeinstallprompt', (event) => { event.preventDefault(); //logic });``` in order to hide/show prompt to install PWA. It seems right now there is no solution for it.

**Comment by joshshipton** [comment]
I'm doing the same thing in a +page.svelte, it fires haphazardly around 5% of the time. Not sure what the solution is right now.

**Comment by teemingc** [comment]
I've created an issue from this discussion here https://github.com/sveltejs/kit/issues/11877
